lib/libc/picolibc/assert.c

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright © 2021, Keith Packard <keithp@keithp.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <zephyr/posix/sys/stat.h>
+#include <sys/time.h>
+#include <zephyr/arch/cpu.h>
+#include <zephyr/linker/linker-defs.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/errno_private.h>
+#include <zephyr/sys/libc-hooks.h>
+#include <zephyr/internal/syscall_handler.h>
+#include <zephyr/app_memory/app_memdomain.h>
+#include <zephyr/init.h>
+#include <zephyr/sys/sem.h>
+#include <zephyr/logging/log.h>
+#ifdef CONFIG_MMU
+#include <zephyr/kernel/mm.h>
+#endif
```

Comment:
Since the same headers are included multiple times by local .c files, would it make sense to put them into a local / private .h file?

The same comment here and elsewhere

Reply:
alternatively, figure out which headers each file needs. You can probably tell I pulled the original file apart and left every single original header in all of the resulting bits. I'll explore which option looks neater in the end. Thanks for taking a look at this series!

Reply:
I trimmed down the include lists so that each file only included what it needed. The results look a lot nicer.